client/ayon_harmony/api/pipeline.py

```diff
@@ -28,6 +42,53 @@
 INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
 
 
+class HarmonyHost(HostBase, IWorkfileHost, ILoadHost, IPublishHost):
+    name = "harmony"
+
+    def install(self):
+        """Install Pype as host config."""
+        print("Installing AYON Harmony Host ...")
+
+        pyblish.api.register_host("harmony")
+        pyblish.api.register_plugin_path(PUBLISH_PATH)
+        register_loader_plugin_path(LOAD_PATH)
+        register_creator_plugin_path(CREATE_PATH)
+
+        register_event_callback("application.launched", application_launch)
+
+    def uninstall(self):
+        pyblish.api.deregister_plugin_path(PUBLISH_PATH)
+        deregister_loader_plugin_path(LOAD_PATH)
+        deregister_creator_plugin_path(CREATE_PATH)
+
+    def open_workfile(self, filepath):
+        return open_file(filepath)
+
+    def save_workfile(self, filepath=None):
+        return save_file(filepath)
+
+    def work_root(self, session):
+        return work_root(session)
+
+    def get_current_workfile(self):
+        return current_file()
+
+    def workfile_has_unsaved_changes(self):
+        return has_unsaved_changes()
+
+    def get_workfile_extensions(self):
+        return file_extensions()
+
+    def get_containers(self):
+        return ls()
+
+    def get_context_data(self):
+        raise NotImplementedError()
+
+    def update_context_data(self, data, changes):
+        raise NotImplementedError()
```

Comment:
To implement the **Context Data** we need to decide where to store this data in the workfile. The context data is the publisher data that is not specific to a particular instance, but to the context. Like whether you have a certain checkbox in the "Context" entry disabled or enabled.

Basically what we need to implement here is a way to store with the workfile some globally accessible JSON data. In essence turning these methods into those calls and we're there.
```suggestion
    def get_context_data(self) -> dict:
        return harmony.get_data("AYON_context")

    def update_context_data(self, data: dict, changes: dict):
        harmony.set_data("AYON_context", data)
```

Note that the `data` is a Python `dict` - so if Harmony only can store a `str` value then you may need to encode/decode it to `json` using e.g. encoding `data: str = json.dumps(data)` and decoding `data: dict = json.loads(data)`

Reply:
Implemented with https://github.com/ynput/ayon-harmony/pull/17/commits/569fc792219f980dbf7ab090dd2333fcf4cbd062